# Process
 
 ### Data Cleaning
 In this phase the data will be uploaded, origanized, cleaned and ready for analysis. 

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import datetime as dt
import seaborn as sns

pd.options.display.max_columns = None

In [2]:
# Reading files 

# Daily Datasets
daily_activity  = pd.read_csv("/Users/gurilyall/Downloads/Analysis Projects/4. bellabeat_raw_data/csv_files/dailyActivity_merged.csv")
daily_calories   = pd.read_csv("/Users/gurilyall/Downloads/Analysis Projects/4. bellabeat_raw_data/csv_files/dailyCalories_merged.csv")
daily_intensities  = pd.read_csv("/Users/gurilyall/Downloads/Analysis Projects/4. bellabeat_raw_data/csv_files/dailyIntensities_merged.csv")
daily_steps  = pd.read_csv("/Users/gurilyall/Downloads/Analysis Projects/4. bellabeat_raw_data/csv_files/dailySteps_merged.csv")

# Other datasets
heartrate_seconds  = pd.read_csv("/Users/gurilyall/Downloads/Analysis Projects/4. bellabeat_raw_data/csv_files/heartrate_seconds_merged.csv")
sleep_log  = pd.read_csv("/Users/gurilyall/Downloads/Analysis Projects/4. bellabeat_raw_data/csv_files/sleepDay_merged.csv")
weight_log  = pd.read_csv("/Users/gurilyall/Downloads/Analysis Projects/4. bellabeat_raw_data/csv_files/weightLogInfo_merged.csv")
print("successful")

successful


In [3]:
# Checking column names of all the dataframes

print("Daily Activity\n",daily_activity.columns,
      "\n\nDaily Calories\n",daily_calories.columns,
      "\n\nDaily Intensities\n",daily_intensities.columns,
      "\n\nDaily Steps\n",daily_steps.columns,
      "\n\nHeart Rate\n",heartrate_seconds.columns,
      "\n\nSleep Log\n",sleep_log.columns)

Daily Activity
 Index(['Id', 'ActivityDate', 'TotalSteps', 'TotalDistance', 'TrackerDistance',
       'LoggedActivitiesDistance', 'VeryActiveDistance',
       'ModeratelyActiveDistance', 'LightActiveDistance',
       'SedentaryActiveDistance', 'VeryActiveMinutes', 'FairlyActiveMinutes',
       'LightlyActiveMinutes', 'SedentaryMinutes', 'Calories'],
      dtype='object') 

Daily Calories
 Index(['Id', 'ActivityDay', 'Calories'], dtype='object') 

Daily Intensities
 Index(['Id', 'ActivityDay', 'SedentaryMinutes', 'LightlyActiveMinutes',
       'FairlyActiveMinutes', 'VeryActiveMinutes', 'SedentaryActiveDistance',
       'LightActiveDistance', 'ModeratelyActiveDistance',
       'VeryActiveDistance'],
      dtype='object') 

Daily Steps
 Index(['Id', 'ActivityDay', 'StepTotal'], dtype='object') 

Heart Rate
 Index(['Id', 'Time', 'Value'], dtype='object') 

Sleep Log
 Index(['Id', 'SleepDay', 'TotalSleepRecords', 'TotalMinutesAsleep',
       'TotalTimeInBed'],
      dtype='object')


lets check the size, shape & datatypes of all the datasets.

In [4]:
# checking data shape, size and data types
print('Activity Size:', daily_activity.size, 
      "\nactivity Shape:" , daily_activity.shape, 
      '\nData types:\n',daily_activity.dtypes )

print(f'{"-"*50}\nCalories Size:', daily_calories.size, 
      "\ncalories Shape:" , daily_calories.shape, 
      '\nData types:\n',daily_calories.dtypes )

print(f'{"-"*50}\nData Intensities Size:', daily_intensities.size, 
      "\ndata Shape:" , daily_intensities.shape, 
      '\nData types:\n',daily_intensities.dtypes )

print(f'{"-"*50}\nSteps Size:', daily_steps.size, 
      "\nsteps Shape:" , daily_steps.shape, 
      '\nData types:\n',daily_steps.dtypes )

print(f'{"-"*50}\nHeart Rate Size:', heartrate_seconds.size,
      "\nHeart Rate Shape:", heartrate_seconds.shape,
      '\nData types:\n',heartrate_seconds.dtypes )

print(f'{"-"*50}\nSleep Log Size:', sleep_log.size,
      "\nSleep Log Shape:", sleep_log.shape,
      '\nData types:\n',sleep_log.dtypes )

print(f'{"-"*50}\nWeight Log Size:', weight_log.size,
      "\nWeight Log Shape:", weight_log.shape,
      '\nData types:\n',weight_log.dtypes )

Activity Size: 14100 
activity Shape: (940, 15) 
Data types:
 Id                            int64
ActivityDate                 object
TotalSteps                    int64
TotalDistance               float64
TrackerDistance             float64
LoggedActivitiesDistance    float64
VeryActiveDistance          float64
ModeratelyActiveDistance    float64
LightActiveDistance         float64
SedentaryActiveDistance     float64
VeryActiveMinutes             int64
FairlyActiveMinutes           int64
LightlyActiveMinutes          int64
SedentaryMinutes              int64
Calories                      int64
dtype: object
--------------------------------------------------
Calories Size: 2820 
calories Shape: (940, 3) 
Data types:
 Id              int64
ActivityDay    object
Calories        int64
dtype: object
--------------------------------------------------
Data Intensities Size: 9400 
data Shape: (940, 10) 
Data types:
 Id                            int64
ActivityDay                  object
Seden

We will count the total number of participants by their unique IDs and analyze the frequency of their participation. By checking the value counts, we can assess the level of engagement and determine whether participants were actively involved in the dataset.

In [5]:
# checking the number of participants in each dataset
print("daily_activity Participants:", daily_activity.Id.nunique(),
      "\ndaily_calories Participants:", daily_calories.Id.nunique(),
      "\ndaily_intensities Participants:", daily_intensities.Id.nunique(),
      "\ndaily_steps Participants:", daily_steps.Id.nunique(),
      "\nHeart Rate Participants:", heartrate_seconds.Id.nunique(),
      "\nSleep Log Participants:", sleep_log.Id.nunique(),
      "\nWeight Log Participants:", weight_log.Id.nunique())

daily_activity Participants: 33 
daily_calories Participants: 33 
daily_intensities Participants: 33 
daily_steps Participants: 33 
Heart Rate Participants: 14 
Sleep Log Participants: 24 
Weight Log Participants: 8


The output shows, `Sleep Log`,`Heart Rate` and `weight_log` have `24`,`14`,`8`  different participants log-ins . All the other datasets have `33`  log-ins from 33 participants.

In the next step we will check the amounts of logins per dataset to check the participation of each person.

In [12]:
pd.set_option('display.width', 1000)  
pd.set_option('display.max_columns', None)  
# checking login(s) for each participant by ID

comparison = pd.concat([
    daily_activity['Id'].value_counts().rename('Activity'),
    daily_calories['Id'].value_counts().rename('Calories'),
    daily_intensities['Id'].value_counts().rename('Intensities'),
    daily_steps['Id'].value_counts().rename('Steps'),
    heartrate_seconds['Id'].value_counts().rename('Heartrate'),
    sleep_log['Id'].value_counts().rename('SleepLog'),
    weight_log['Id'].value_counts().rename('WeightLog')
], axis=1)

print(f"{" "*20}Login(s) for each participant by ID\n\n",comparison)

                    Login(s) for each participant by ID

             Activity  Calories  Intensities  Steps  Heartrate  SleepLog  WeightLog
Id                                                                                
1503960366        31        31           31     31        NaN      25.0        2.0
4319703577        31        31           31     31        NaN      26.0        2.0
8583815059        31        31           31     31        NaN       NaN        NaN
8378563200        31        31           31     31        NaN      32.0        NaN
8053475328        31        31           31     31        NaN       3.0        NaN
7086361926        31        31           31     31        NaN      24.0        NaN
6962181067        31        31           31     31   266326.0      31.0       30.0
5553957443        31        31           31     31   255174.0      31.0        NaN
4702921684        31        31           31     31        NaN      28.0        NaN
4558609924        31        3

For the daily datasets the particpation is good  except 1 women with `4` entries. Fewer number of women participated in heartrate_seconds, Sleep_log, weight_log according to that there are alot of `Nan`s in these 3 datasets

In this step, we are examining all the datasets to identify any missing data. This helps us understand the completeness of the datasets and determine how to handle any gaps or inconsistencies in the data moving forward.

In [7]:
# Checking for missing values in each dataset
print("daily_activity Missing Values:", daily_activity.isnull().sum().sum(),  
      "\ndaily_calories Missing Values:", daily_calories.isnull().sum().sum(),  
      "\ndaily_intensities Missing Values:", daily_intensities.isnull().sum().sum(),  
      "\ndaily_steps Missing Values:", daily_steps.isnull().sum().sum(),  
      "\nHeart Rate Missing Values:", heartrate_seconds.isnull().sum().sum(),  
      "\nSleep Log Missing Values:", sleep_log.isnull().sum().sum(),
      "\nWeight Log Missing Values:", weight_log.isnull().sum().sum())  

daily_activity Missing Values: 0 
daily_calories Missing Values: 0 
daily_intensities Missing Values: 0 
daily_steps Missing Values: 0 
Heart Rate Missing Values: 0 
Sleep Log Missing Values: 0 
Weight Log Missing Values: 65


In [13]:
weight_log.isnull().sum()

Id                 0
Date               0
WeightKg           0
WeightPounds       0
Fat               65
BMI                0
IsManualReport     0
LogId              0
dtype: int64

The weight_log dataset contains `67` rows of data, with `65` of them having null values in the `Fat` column. This indicates that most of the data is missing, limiting the insights that can be drawn from the Fat column. Meanwhile, this issue has been forwarded to the Marketing Analysis Team and Bellabeat’s IT Team for further investigation. We will not drop the data and will continue with the analysis.

Checking for Duplicated entries in the datasets.

In [14]:
# checking for duplicates
print("daily_activity Duplicates:", daily_activity.duplicated().sum(),  
      "\ndaily_calories Duplicates:", daily_calories.duplicated().sum(),  
      "\ndaily_intensities Duplicates:", daily_intensities.duplicated().sum(),  
      "\ndaily_steps Duplicates:", daily_steps.duplicated().sum(),  
      "\nHeart Rate Duplicates:", heartrate_seconds.duplicated().sum(),  
      "\nSleep Log Duplicates:", sleep_log.duplicated().sum(),
      "\nWeight Log Duplicates:", weight_log.duplicated().sum())  

daily_activity Duplicates: 0 
daily_calories Duplicates: 0 
daily_intensities Duplicates: 0 
daily_steps Duplicates: 0 
Heart Rate Duplicates: 0 
Sleep Log Duplicates: 3 
Weight Log Duplicates: 0


As per the output there are `3` duplicated entries in the sleep log dataset. In the next step, dropping the duplicated values. 

In [15]:
# dropping duplicated and null data entries
sleep_log.drop_duplicates(inplace=True)
print("Sleep Log Duplicates:", sleep_log.duplicated().sum())  


Sleep Log Duplicates: 0


# Analyze

### Data Transformation
In this Phase, we will transform the data using various methods, such as changing data types, renaming columns, merging datasets, and creating new columns or aliases. These changes will be made in a way that preserves the original data, allowing it to be accessed or reviewed if needed.

In [16]:
# renaming the columns
daily_activity = daily_activity.rename(columns = {'ActivityDate':'ActivityDay'})


In [17]:
# mering all the daily datasets into one
df1 = pd.merge(pd.merge(pd.merge(daily_activity, daily_calories, on=['Id', 'ActivityDay'], how='outer'), daily_intensities, on=['Id', 'ActivityDay'], how='outer'), daily_steps, on=['Id', 'ActivityDay'], how='outer')

print("Missing Data: ",df1.isnull().sum().sum(),
      "\nDataset Shape: ", df1.shape,
      "\nDataset Size: ", df1.size)

Missing Data:  0 
Dataset Shape:  (940, 25) 
Dataset Size:  23500


The dataset shape indicates `940` rows and `25` columns, which aligns with the combined total from the initial four datasets

Changing the format of the of all the date for all the datasets

In [18]:
# Daily dataset date formatting
df1['ActivityDay'] = pd.to_datetime(df1['ActivityDay'], format="%m/%d/%Y")

# Heart rate dataset date and time formatting
heart = heartrate_seconds.copy()
heart['Date'] = pd.to_datetime(heart['Time'], format="%m/%d/%Y %I:%M:%S %p")

# Weight log dataset date formatting
weight = weight_log.copy()
weight['Date'] = pd.to_datetime(weight['Date'], format="%m/%d/%Y %I:%M:%S %p")

# Sleep log dataset date formatting
sleep = sleep_log.copy()
sleep['SleepDay'] = pd.to_datetime(sleep['SleepDay'], format="%m/%d/%Y %I:%M:%S %p")

print("Daily Dataset", df1['ActivityDay'].dtype,
      "\nHeart Rate Dataset", heart['Date'].dtype,
      "\nWeight Log Dataset", weight['Date'].dtype,
      "\nSleep Log Dataset", sleep['SleepDay'].dtype)

Daily Dataset datetime64[ns] 
Heart Rate Dataset datetime64[ns] 
Weight Log Dataset datetime64[ns] 
Sleep Log Dataset datetime64[ns]


Making a copy to work with, so the original data available as a back-up. Creating a new column `DayOfTheWeek` by extrating data from the `ActivityDay`. With the data available for 2 month `DayOfTheWeek` would help to understand the pattern of the fitbit data.

In [19]:
# reindex function to rearrange the data
df_2= df1.copy()

# print 1st 5 rows to confirm
df_2['DayOfTheWeek'] = df_2['ActivityDay'].dt.day_name()
df_2.DayOfTheWeek.value_counts()
# df_2.info()

heart['DayOfTheWeek'] = heart['Date'].dt.day_name()
heart.DayOfTheWeek.value_counts()

DayOfTheWeek
Tuesday      452381
Wednesday    399232
Thursday     359146
Friday       357594
Saturday     338394
Monday       289764
Sunday       287147
Name: count, dtype: int64

To save time when checking each row and column for duplicates, a function called check_columns_match was developed. This function compares data at each index for columns with the same name. To keep the process simple, columns ending with `_x` were assigned to the `x` variable, while those ending with `_y` were assigned to the `y` variable.


The function outputs whether the columns match or mismatch. If the data matches, the duplicate column will be dropped to reduce redundancy. Just to make sure the function is working properly  2 different columns were added to the variables.

In [20]:
# to check if the data in these columns are same 
x = ['VeryActiveMinutes_x', 'FairlyActiveMinutes_x', 'LightlyActiveMinutes_x', 'SedentaryMinutes_x','VeryActiveDistance_x', 'ModeratelyActiveDistance_x', 'LightActiveDistance_x', 'SedentaryActiveDistance_x', 'Calories_x','TotalSteps','TotalSteps']
y = ['VeryActiveMinutes_y', 'FairlyActiveMinutes_y', 'LightlyActiveMinutes_y', 'SedentaryMinutes_y','VeryActiveDistance_y', 'ModeratelyActiveDistance_y', 'LightActiveDistance_y', 'SedentaryActiveDistance_y', 'Calories_y','StepTotal','TrackerDistance']
    

def check_columns_match(df, col_x_list, col_y_list):
    for col_x, col_y in zip(col_x_list, col_y_list):
        match = True
        for i in range(len(df)):
            if df[col_x][i] != df[col_y][i]:
                match = False
                print(f"\nMismatch in columns '{col_x}' and '{col_y}' at index {i}")
                break

        if match:
            print(f"Match: All values in columns '{col_x}' and '{col_y}' are the same.")
        else:
            print(f"Mismatch: Not all values in columns '{col_x}' and '{col_y}' are the same.")


check_columns_match(df_2, x, y)

Match: All values in columns 'VeryActiveMinutes_x' and 'VeryActiveMinutes_y' are the same.
Match: All values in columns 'FairlyActiveMinutes_x' and 'FairlyActiveMinutes_y' are the same.
Match: All values in columns 'LightlyActiveMinutes_x' and 'LightlyActiveMinutes_y' are the same.
Match: All values in columns 'SedentaryMinutes_x' and 'SedentaryMinutes_y' are the same.
Match: All values in columns 'VeryActiveDistance_x' and 'VeryActiveDistance_y' are the same.
Match: All values in columns 'ModeratelyActiveDistance_x' and 'ModeratelyActiveDistance_y' are the same.
Match: All values in columns 'LightActiveDistance_x' and 'LightActiveDistance_y' are the same.
Match: All values in columns 'SedentaryActiveDistance_x' and 'SedentaryActiveDistance_y' are the same.
Match: All values in columns 'Calories_x' and 'Calories_y' are the same.
Match: All values in columns 'TotalSteps' and 'StepTotal' are the same.

Mismatch in columns 'TotalSteps' and 'TrackerDistance' at index 0
Mismatch: Not all va

All columns in the `x` variable `match` with those in the `y`. Therefore, to improve efficiency and reduce redundancy, all columns ending with `_y` will be dropped.

In [21]:
# deleting duplicate columns
df_2 = df_2.drop(columns = ['VeryActiveMinutes_y', 'FairlyActiveMinutes_y', 'LightlyActiveMinutes_y', 'SedentaryMinutes_y', 
        'VeryActiveDistance_y', 'ModeratelyActiveDistance_y', 'LightActiveDistance_y', 'SedentaryActiveDistance_y', 'Calories_y','StepTotal'])
df_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 16 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   Id                          940 non-null    int64         
 1   ActivityDay                 940 non-null    datetime64[ns]
 2   TotalSteps                  940 non-null    int64         
 3   TotalDistance               940 non-null    float64       
 4   TrackerDistance             940 non-null    float64       
 5   LoggedActivitiesDistance    940 non-null    float64       
 6   VeryActiveDistance_x        940 non-null    float64       
 7   ModeratelyActiveDistance_x  940 non-null    float64       
 8   LightActiveDistance_x       940 non-null    float64       
 9   SedentaryActiveDistance_x   940 non-null    float64       
 10  VeryActiveMinutes_x         940 non-null    int64         
 11  FairlyActiveMinutes_x       940 non-null    int64         

I have created a new column, `total_mins` & `total_hours`, to aggregate the total time spent across all activity phases, providing a comprehensive view of the overall activity duration.

In [22]:
#total mins column
df_2['total_mins'] = df_2['VeryActiveMinutes_x'] + df_2['FairlyActiveMinutes_x'] + df_2['LightlyActiveMinutes_x'] + df_2['SedentaryMinutes_x'] 

#total hours column
df_2['total_hours'] =round(df_2['total_mins']/60)

df_2[['total_mins','total_hours','DayOfTheWeek']].head(10)

total_mins  total_hours DayOfTheWeek
0        1094         18.0      Tuesday
1        1033         17.0    Wednesday
2        1440         24.0     Thursday
3         998         17.0       Friday
4        1040         17.0     Saturday
5         761         13.0       Sunday
6        1440         24.0       Monday
7        1120         19.0      Tuesday
8        1063         18.0    Wednesday
9        1076         18.0     Thursday

creating new column in the sleep dataset to check the awake time in bed

In [23]:
sleep['inBed_awake'] = sleep.TotalTimeInBed - sleep.TotalMinutesAsleep

print("Total Participants:", sleep.Id.nunique())
sleep.head(5)

Total Participants: 24


Id   SleepDay  TotalSleepRecords  TotalMinutesAsleep  TotalTimeInBed  inBed_awake
0  1503960366 2016-04-12                  1                 327             346           19
1  1503960366 2016-04-13                  2                 384             407           23
2  1503960366 2016-04-15                  1                 412             442           30
3  1503960366 2016-04-16                  2                 340             367           27
4  1503960366 2016-04-17                  1                 700             712           12

Checking mean awake time in bed by Id

In [24]:
# checking mean sleep and awake time for each id
awake_mean_df = sleep[["Id","inBed_awake","TotalMinutesAsleep"]]

awake_mean = awake_mean_df.groupby("Id")['inBed_awake'].mean().reset_index()
sleep_mean = awake_mean_df.groupby("Id")['TotalMinutesAsleep'].mean().reset_index()

sleep_mean_df = pd.merge(awake_mean, sleep_mean, on="Id")

sleep_mean_df.columns = ["Id","mean_inBed_awake","mean_sleep"]
sleep_mean_df.head(4)

Id  mean_inBed_awake  mean_sleep
0  1503960366             22.92      360.28
1  1644430081             52.00      294.00
2  1844505072            309.00      652.00
3  1927972279             20.80      417.00

In the weight dataset, a new column Month was created to separate the data by month, allowing for a more detailed analysis of variations in the weightKg column.

In [36]:

# Extract the month from the 'Date' column and create a new 'Month' column
weight['Month'] = weight['Date'].dt.month
weight.Month.head()

april_raw = weight[weight['Month'] == 4]
may_raw = weight[weight['Month'] == 5]

# print(april['Id'].unique())
# print(may['Id'].unique())

april = april_raw[april_raw['Id'].isin(may_raw['Id'])]  # Full rows from April
may = may_raw[may_raw['Id'].isin(april_raw['Id'])] 
print('April Id count',april.Id.nunique(),
        '\nMay Id count',may.Id.nunique())


April Id count 5 
May Id count 5


In [26]:
april_stats = april[["BMI", "WeightKg"]].describe()
may_stats = may[["BMI", "WeightKg"]].describe()

# Combine side by side for comparison
comparison = pd.concat([april_stats, may_stats], axis=1, keys=['April', 'May'])

# Display result
print(comparison)

           April                   May           
             BMI   WeightKg        BMI   WeightKg
count  38.000000  38.000000  25.000000  25.000000
mean   24.847895  72.118422  24.894000  70.260000
std     1.170937  11.639561   1.340469  10.841549
min    21.450001  56.700001  21.690001  57.299999
25%    23.969999  61.425001  24.000000  61.500000
50%    24.840000  66.099998  24.350000  62.400002
75%    25.559999  85.099998  25.559999  84.400002
max    27.459999  85.800003  27.379999  85.500000


Going in depth to check if any  change occured in the of the participating women

In [40]:
# Checking weight and BMI for each id for April and May
april_stats = april.groupby('Id')[["BMI", "WeightKg"]].mean().rename(columns={'BMI': 'April_BMI'})
may_stats = may.groupby('Id')[["BMI", "WeightKg"]].mean().rename(columns={'BMI': 'May_BMI'})

combined_stats = pd.concat([april_stats, may_stats], axis=1)

# Display the combined stats
print(combined_stats)


            April_BMI   WeightKg    May_BMI   WeightKg
Id                                                    
2873212765  21.450001  56.700001  21.690001  57.299999
4319703577  27.450001  72.400002  27.379999  72.300003
4558609924  27.355000  70.000000  27.120000  69.399999
6962181067  24.023888  61.544445  24.034167  61.566667
8877689391  25.511250  85.225001  25.438750  84.987501


Over the two-month period of data collection, the weight dataset shows minimal variation in BMI and weight statistics across participants, indicating largely consistent results throughout the period.


In [ ]:
# fucntion for day count
def count_day(day_of_week):
    count = 0
    for day in df_2.DayOfTheWeek:
        if day == day_of_week:
            count += 1
    return count

Mon = count_day("Monday")
Tue = count_day("Tuesday")
Wed = count_day("Wednesday")
Thu = count_day("Thursday")
Fri = count_day("Friday")
Sat = count_day("Saturday")
Sun = count_day("Sunday")

# mean stats 
mean_calories = df_2.Calories_x.mean()
mean_hours = df_2.total_hours.mean()
mean_sedentary = df_2.SedentaryMinutes_x.mean()/60 # converting to minutes
mean_light =df_2. LightlyActiveMinutes_x.mean()/60 # converting to minutes
mean_fairlyActive = df_2.FairlyActiveMinutes_x.mean()
mean_V_active = (df_2.VeryActiveMinutes_x.mean()/60)

#mean distance stats
mean_calories = round(df_2.Calories_x.mean(),2)
mean_distance = round(df_2.TotalDistance.mean(),2)
mean_sedentary_d = round(df_2.SedentaryActiveDistance_x.mean(),2)
mean_light_d =  round(df_2.LightActiveDistance_x.mean(),2) 
mean_moderate_d = round(df_2.ModeratelyActiveDistance_x.mean(),2)
mean_V_active_d = round(df_2.VeryActiveDistance_x.mean(),2)
mean_logged_d =round(df_2.LoggedActivitiesDistance.mean(),2)
mean_tracker_d = round(df_2.TrackerDistance.mean(),2)

# mean  step stats
mean_steps =round(df_2.TotalSteps.mean())
mean_distance = round(df_2.TotalDistance.mean())